api/src/main/java/io/minio/MinioClient.java

```diff
@@ -2383,6 +2386,240 @@ public void copyObject(String bucketName, String objectName, Map<String,String>
+   *
+   * @param bucketName
+   *          Destination Bucket to be created upon compose.
+   * @param objectName
+   *          Destination Object to be created upon compose.
+   * @param sources
+   *          List of Source Objects used to compose Object.
+   * @param headerMap
+   *          User Meta data.
+   * @param sse
+   *          Server Side Encryption.
+   *
+   * @throws InvalidBucketNameException  upon invalid bucket name is given
+   * @throws NoSuchAlgorithmException
+   *           upon requested algorithm was not found during signature calculation
+   * @throws InsufficientDataException  upon getting EOFException while reading given
+   *           InputStream even before reading given length
+   * @throws IOException                 upon connection error
+   * @throws InvalidKeyException
+   *           upon an invalid access key or secret key
+   * @throws NoResponseException         upon no response from server
+   * @throws XmlPullParserException      upon parsing response xml
+   * @throws ErrorResponseException      upon unsuccessful execution
+   * @throws InternalException           upon internal library error
+   * @throws InvalidArgumentException    upon invalid value is passed to a method.
+   * @throws InvalidResponseException    upon a non-xml response from server
+   */
+
+  public void composeObject(String bucketName, String objectName, List<ComposeSource> sources,
+      Map<String,String> headerMap, ServerSideEncryption sse)
+    throws InvalidBucketNameException, NoSuchAlgorithmException, InsufficientDataException, IOException,
+    InvalidKeyException, NoResponseException, XmlPullParserException, ErrorResponseException,
+    InternalException, InvalidArgumentException, InvalidResponseException {
+    if (bucketName == null) {
+      throw new InvalidArgumentException("Destination bucket name cannot be empty");
+    }
+
+    if (objectName == null) {
+      throw new InvalidArgumentException("Destination object name cannot be empty");
+    }
+
+    if (sources.isEmpty()) {
+      throw new InvalidArgumentException("compose sources cannot be empty");
+    }
+
+    checkWriteRequestSse(sse);
+
+    long objectSize = 0;
+    int partsCount = 0;
+    for (int i = 0; i < sources.size(); i++) {
+      ComposeSource src = sources.get(i);
+
+      checkReadRequestSse(src.sse());
+
+      ObjectStat stat = statObject(src.bucketName(), src.objectName(), src.sse());
+      src.buildHeaders(stat.length(), stat.etag());
+
+      if (i != 0 && src.headers().containsKey("x-amz-meta-x-amz-key")) {
+        throw new InvalidArgumentException("Client side encryption is not supported for more than one source");
+      }
+
+      long size = stat.length();
+      if (src.length() != null) {
+        size = src.length();
+      } else {
+        if (src.offset() != null) {
+          size -= src.offset();
+        }
+      }
+
+      if (size < MIN_MULTIPART_SIZE && sources.size() != 1 && i != (sources.size() - 1)) {
```

Comment:
@sinhaashish 
Could you explain this line for me, specifically the 2nd and 3rd conditions?

Reply:
Yes @ebozduman  . Its similar to `minio-go`  implementation [here ](https://github.com/minio/minio-go/blob/e11840609c348b5d1e46185b7f6e98ce6e6bc313/api-compose-object.go#L395)